packages/grafana-ui/src/components/TimeSeries/utils.ts

```diff
@@ -259,10 +261,25 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn<{ sync: DashboardCursor
   return builder;
 };
 
-export function getNamesToFieldIndex(frame: DataFrame): Map<string, number> {
-  const names = new Map<string, number>();
+export function getNamesToFieldIndex(
+  frame: DataFrame,
+  allFrames?: DataFrame[]
+): { alignedNames: Map<string, number>; originNames: Map<string, number> } {
+  const alignedNames = new Map<string, number>();
+  const originNames = new Map<string, number>();
   for (let i = 0; i < frame.fields.length; i++) {
-    names.set(getFieldDisplayName(frame.fields[i], frame), i);
+    const origin = frame.fields[i].state?.origin;
+    if (allFrames && origin) {
+      originNames.set(
+        calculateFieldDisplayName(
```

Comment:
In this case, since we know `frame` exists, this means that `allFrames` exist too. I think there's no need for using the `calculateFieldDisplayName` but just getFieldDisplayName and utilize the cache if it was calculated for allFrames field already. This would save us exposing the `calculateFieldDisplayName` function.

Reply:
We can't run the getFieldDisplayName over the dataframe as it will change the state that is by reference. Before this PR when you add the fillBelowTo config property `A RH` changes to `RH A` and `B Temp` changes to `Temp B` everywhere in the graph. I previously had a way to disable the updating of the cache, but all it really did was make getFieldDisplayName into calculateFieldDisplayName.

Reply:
The issue with changing from A RH to RH A was caused by calling the getDisplayName on a field (with the already cached state) against aligned data frame. But since the aligned data frame has indexes to both frame and field within the original data frame, this should no longer be an issue if you call getFieldDisplayName on **original** field and **original** frame.